### DISCUSSION TITLE
Importer for CAD files

### DISCUSSION THREAD

**Discussion by eddychen06** [sofa-framework/sofa#3171] [Ideas / Suggestions]
Having a way to import CAD assembly or part files into a simulation would make it convenient for those using CAD programs to design models. For objects with multiple parts, this could remove the need to convert separate parts into different mesh files.

**Comment by epernod** [comment]
Hi,

could you detail the file format you would like to support or point to the official online doc.
Would the library assimp do the job? 

best,

**Reply by eddychen06** [reply]
I think the library assimp might be able to do the job. I use AutoDesk Inventor and the part files are in the format .ipt and assembly files are in the format .iam. However, I am not too sure about the file types in other CAD programs. 

This link goes over the file types in Inventor: https://knowledge.autodesk.com/support/inventor/learn-explore/caas/CloudHelp/cloudhelp/2021/ENU/Inventor-Help/files/GUID-94B779C0-6B2B-499A-A4F9-2E4BAB49712F-htm.html.

**Reply by epernod** [reply]
looking at assimp documentation, I'm not sure it handle those format: https://assimp-docs.readthedocs.io/en/latest/about/introduction.html
